.travis.yml

```diff
@@ -1,26 +1,37 @@
-dist: trusty
 language: node_js
 node_js:
-  - 8
-  
+- '8'
+- '10'
+
 os:
-  - osx
-  - linux
+- linux
+- osx
+
+notifications:
+  email: false
+
 addons:
   apt:
     packages:
-      - icnsutils
-      - graphicsmagick
-      - xz-utils
-      - xvfb
+    - xvfb
+
 env:
   - DISPLAY=":99.0"
+
 install:
   - Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
   - npm install
+
+# Trigger deploy in PR
+# after_success:
+#   - npm run dist:publish
+
 deploy:
   provider: script
-  script: npm run dist
-  on:
-    branch: master
```

Comment:
There is some general problem here, actually there are a few:

- `deploy` jobs are always skipped in pull requests, so whatever you configure here, it will not run in a pull request. 
- `npm install` and `npm test` seem to be always run independent from this.

It's very difficult to test this correctly as long as I'm in a pull request. I think this needs some adjustments and test, when it's available in the final branch. That's why a couple of things are commented out.